internal/cmd/byok/command_register.go

```diff
@@ -0,0 +1,222 @@
+	"Effect" : "Allow",
+	"Principal" : {
+		"AWS" : [{{range $i, $e := .}}{{if $i}},{{end}}
+			"{{$e}}"{{end}}
+		]
+	},
+	"Action" : [ "kms:CreateGrant", "kms:ListGrants", "kms:RevokeGrant" ],
+	"Resource" : "*"
+}`))
+
+const keyVaultCryptoServiceEncryptionUser = "e147488a-f6f5-4113-8e2d-b22465e65bf6"
+const keyVaultReader = "21090545-7ca7-4776-b22c-e363652d74d2"
+
+func (c *command) newRegisterCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "register <key>",
+		Short: "Register a self-managed key in Confluent Cloud.",
+		Args:  cobra.ExactArgs(1),
+		RunE:  c.register,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Register a new self-managed encryption key for AWS:",
+				Code: `confluent byok register "arn:aws:kms:us-west-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab"`,
+			},
+			examples.Example{
+				Text: "Register a new self-managed encryption key for Azure:",
+				Code: `confluent byok register "https://a-vault.vault.azure.net/keys/a-key/00000000000000000000000000000000" --tenantid "00000000-0000-0000-0000-000000000000" --key-vault-id "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/a-resourcegroups/providers/Microsoft.KeyVault/vaults/a-vault"`,
+			},
+		),
+	}
+
+	cmd.Flags().String("key-vault-id", "", "The ID of the Azure Key Vault where the key is stored.")
+	cmd.Flags().String("tenant-id", "", "The ID of the Azure Active Directory tenant that the key vault belongs to.")
+	cmd.MarkFlagsRequiredTogether("key-vault-id", "tenant-id")
+	pcmd.AddOutputFlag(cmd)
+
+	return cmd
+}
+
+func (c *command) registerAWS(cmd *cobra.Command, keyString string) (*byokv1.ByokV1Key, error) {
+	keyReq := byokv1.ByokV1Key{}
+
+	keyReq.Key = &byokv1.ByokV1KeyKeyOneOf{
+		ByokV1AwsKey: &byokv1.ByokV1AwsKey{
+			KeyArn: keyString,
+			Kind:   "AwsKey",
+		},
+	}
+
+	keyResp, httpResp, err := c.V2Client.CreateByokKey(keyReq)
+	if err != nil {
+		return nil, errors.CatchCCloudV2Error(err, httpResp)
+	}
+
+	return &keyResp, nil
+}
+
+func (c *command) registerAzure(cmd *cobra.Command, keyString string) (*byokv1.ByokV1Key, error) {
+	keyReq := byokv1.ByokV1Key{}
+
+	keyVaultID, err := cmd.Flags().GetString("key-vault-id")
+	if err != nil {
+		return nil, err
+	}
+	tenantID, err := cmd.Flags().GetString("tenant-id")
+	if err != nil {
+		return nil, err
+	}
+
+	keyReq.Key = &byokv1.ByokV1KeyKeyOneOf{
+		ByokV1AzureKey: &byokv1.ByokV1AzureKey{
+			KeyId:      keyString,
+			KeyVaultId: keyVaultID,
+			TenantId:   tenantID,
+			Kind:       "AzureKey",
+		},
+	}
+
+	keyResp, httpResp, err := c.V2Client.CreateByokKey(keyReq)
+	if err != nil {
+		return nil, errors.CatchCCloudV2Error(err, httpResp)
+	}
+
+	return &keyResp, err
+
+}
+
+func (c *command) register(cmd *cobra.Command, args []string) error {
+	keyString := args[0]
+	var err error
+
+	/* outputFormat, err := cmd.Flags().GetString(output.FlagName)
+	if err != nil {
+		return err
+	} */
+
+	var postCreationStepInstructions string
```

Comment:
No need to define this separately. Just use `:=` when it's set.

Reply:
Fixed.